src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let backupObj = Object.assign({}, state);
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        backupObj = { ...backupObj };
```

Comment:
do you really need to assign it on each case?
i don't think so - you just mutate it on each `case`, but then the only place you need a copy of it - is on `.push` - see the next comment